### ISSUE TITLE
[Bug] [Windows] bazel test python/ray/serve:test_handle times out

### ISSUE THREAD

**Issue by mattip** [ray-project/ray#21106]
### Search before asking

- [X] I searched the [issues](https://github.com/ray-project/ray/issues) and found no similar issues.

### Ray Component

Ray Serve

### What happened + What you expected to happen

When running the test_handle tests on windows via pytest, the tests pass
```
>python -m pytest python\ray\serve\tests\test_handle.py
```
<details><summary>test output</summary>

```
================ test session starts =======================
platform win32 -- Python 3.8.10, pytest-5.4.3, py-1.10.0, pluggy-0.13.1
rootdir: C:\oss\ray-git\python
plugins: asyncio-0.15.1, lazy-fixture-0.6.3, rerunfailures-10.2, timeout-1.4.2
collected 12 items

python\ray\serve\tests\test_handle.py ............                                                               [100%]

======== warnings summary ==============
ray/serve/tests/test_handle.py: 19 tests with warnings
  C:\oss\ray-git\python\ray\serve\router.py:73: DeprecationWarning: The loop argument is deprecated since Python 3.8, and scheduled for removal in Python 3.10.
    self.config_updated_event = asyncio.Event(loop=event_loop)

-- Docs: https://docs.pytest.org/en/latest/warnings.html
==================== 12 passed, 19 warnings in 58.42s ================
```

</details>

When running them via bazel, there is a timeout
```
>bazel test python/ray/serve:test_handle
DEBUG: bazel/ray_deps_setup.bzl:66:14: No implicit mirrors used because urls were explicitly provided
INFO: Analyzed target //python/ray/serve:test_handle (0 packages loaded, 0 targets configured).
INFO: Found 1 test target...

TIMEOUT: //python/ray/serve:test_handle (Summary)
      C:/users/matti/_bazel_matti/jnwh2qma/execroot/com_github_ray_project_ray/bazel-out/x64_windows-opt/testlogs/python/ray/serve/test_handle/test.log
Target //python/ray/serve:test_handle up-to-date:
  bazel-bin/python/ray/serve/test_handle.exe
  bazel-bin/python/ray/serve/test_handle.zip
INFO: Elapsed time: 60.897s, Critical Path: 60.19s
INFO: 2 processes: 2 local.
INFO: Build completed, 1 test FAILED, 2 total actions
//python/ray/serve:test_handle                                          TIMEOUT in 60.1s
  C:/users/matti/_bazel_matti/jnwh2qma/execroot/com_github_ray_project_ray/bazel-out/x64_windows-opt/testlogs/python/ray/serve/test_handle/test.log

INFO: Build completed, 1 test FAILED, 2 total actions

>python -c "print(open('C:/users/matti/_bazel_matti/jnwh2qma/execroot/com_github_ray_project_ray/bazel-out/x64_windows-opt/testlogs/python/ray/serve/test_handle/test.log').read())"

Executing tests from //python/ray/serve:test_handle
-----------------------------------------------------------------------------
============================= test session starts =============================
platform win32 -- Python 3.8.10, pytest-5.4.3, py-1.10.0, pluggy-0.13.1 -- C:\temp\ray_dev\Scripts\python.exe
cachedir: .pytest_cache
rootdir: C:\Users\Matti\AppData\Local\Temp\Bazel.runfiles_pptquq0i\runfiles\com_github_ray_project_ray
plugins: asyncio-0.15.1, lazy-fixture-0.6.3, rerunfailures-10.2, timeout-1.4.2
collecting ... collected 12 items

::test_async_handle_serializable 2021-12-15 08:19:29,090        INFO services.py:1370 -- View the Ray dashboard at http://127.0.0.1:8265
(ServeController
 pid=1924) 2021-12-15 08:19:32,313      INFO checkpoint_path.py:16 -- Using RayInternalKVStore for controller checkpoint and recovery.

(ServeController
 pid=1924) 2021-12-15 08:19:32,316      INFO http_state.py:98 -- Starting HTTP proxy with name 'SERVE_CONTROLLER_ACTOR:SERVE_PROXY_ACTOR-node:127.0.0.1-0' on node 'node:127.0.0.1-0' listening on '127.0.0.1:8000'

2021-12-15 08:19:32,754 INFO api.py:463 -- Started detached Serve instance in namespace 'default_test_namespace'.
2021-12-15 08:19:32,760 INFO api.py:242 -- Updating deployment 'f'. component=serve deployment=f
(HTTPProxyActor
 pid=6848) INFO:     Started server process [6848]

(ServeController
 pid=1924) 2021-12-15 08:19:32,862      INFO deployment_state.py:920 -- Adding 1 replicas to deployment 'f'. component=serve deployment=f

2021-12-15 08:19:33,316 INFO api.py:249 -- Deployment 'f' is ready at `http://127.0.0.1:8000/f`. component=serve deployment=f
PASSED2021-12-15 08:19:33,861   ERROR long_poll.py:155 -- The event loop is closed, shutting down long poll client.

::test_sync_handle_serializable (ServeController
 pid=1924) 2021-12-15 08:19:33,858      INFO deployment_state.py:940 -- Removing 1 replicas from deployment 'f'. component=serve deployment=f

2021-12-15 08:19:35,979 INFO api.py:242 -- Updating deployment 'f'. component=serve deployment=f
(ServeController
 pid=1924) 2021-12-15 08:19:36,073      INFO deployment_state.py:920 -- Adding 1 replicas to deployment 'f'. component=serve deployment=f

2021-12-15 08:19:36,533 INFO api.py:249 -- Deployment 'f' is ready at `http://127.0.0.1:8000/f`. component=serve deployment=f
PASSED
::test_sync_handle_in_thread (ServeController
 pid=1924) 2021-12-15 08:19:36,994      INFO deployment_state.py:940 -- Removing 1 replicas from deployment 'f'. component=serve deployment=f

2021-12-15 08:19:39,045 INFO api.py:242 -- Updating deployment 'f'. component=serve deployment=f
(ServeController
 pid=1924) 2021-12-15 08:19:39,140      INFO deployment_state.py:920 -- Adding 1 replicas to deployment 'f'. component=serve deployment=f

2021-12-15 08:19:39,268 INFO api.py:249 -- Deployment 'f' is ready at `http://127.0.0.1:8000/f`. component=serve deployment=f
PASSED
::test_handle_in_endpoint (ServeController
 pid=1924) 2021-12-15 08:19:39,386      INFO deployment_state.py:940 -- Removing 1 replicas from deployment 'f'. component=serve deployment=f

2021-12-15 08:19:41,505 INFO api.py:242 -- Updating deployment 'Endpoint1'. component=serve deployment=Endpoint1
(ServeController
 pid=1924) 2021-12-15 08:19:41,602      INFO deployment_state.py:920 -- Adding 1 replicas to deployment 'Endpoint1'. component=serve deployment=Endpoint1

2021-12-15 08:19:42,083 INFO api.py:249 -- Deployment 'Endpoint1' is ready at `http://127.0.0.1:8000/Endpoint1`. component=serve deployment=Endpoint1
2021-12-15 08:19:42,091 INFO api.py:242 -- Updating deployment 'Endpoint2'. component=serve deployment=Endpoint2
(ServeController
 pid=1924) 2021-12-15 08:19:42,193      INFO deployment_state.py:920 -- Adding 1 replicas to deployment 'Endpoint2'. component=serve deployment=Endpoint2

2021-12-15 08:19:42,641 INFO api.py:249 -- Deployment 'Endpoint2' is ready at `http://127.0.0.1:8000/Endpoint2`. component=serve deployment=Endpoint2
PASSED
::test_handle_inject_starlette_request (Endpoint2
 pid=2796) 2021-12-15 08:19:42,597      WARNING api.py:308 -- You are retrieving a sync handle inside an asyncio loop. Try getting client.get_handle(.., sync=False) to get better performance. Learn more at https://docs.ray.io/en/master/serve/http-servehandle.html#sync-and-async-handles

(ServeController
 pid=1924) 2021-12-15 08:19:42,757      INFO deployment_state.py:940 -- Removing 1 replicas from deployment 'Endpoint1'. component=serve deployment=Endpoint1

(ServeController
 pid=1924) 2021-12-15 08:19:44,903      INFO deployment_state.py:940 -- Removing 1 replicas from deployment 'Endpoint2'. component=serve deployment=Endpoint2

2021-12-15 08:19:47,010 INFO api.py:242 -- Updating deployment 'echo'. component=serve deployment=echo
(ServeController
 pid=1924) 2021-12-15 08:19:47,112      INFO deployment_state.py:920 -- Adding 1 replicas to deployment 'echo'. component=serve deployment=echo

2021-12-15 08:19:48,555 INFO api.py:249 -- Deployment 'echo' is ready at `http://127.0.0.1:8000/echo`. component=serve deployment=echo
2021-12-15 08:19:48,562 INFO api.py:242 -- Updating deployment 'wrapper'. component=serve deployment=wrapper
(ServeController
 pid=1924) 2021-12-15 08:19:48,681      INFO deployment_state.py:920 -- Adding 1 replicas to deployment 'wrapper'. component=serve deployment=wrapper

2021-12-15 08:19:50,120 INFO api.py:249 -- Deployment 'wrapper' is ready at `http://127.0.0.1:8000/wrapper`. component=serve deployment=wrapper
PASSED
::test_handle_option_chaining (wrapper
 pid=9360) 2021-12-15 08:19:50,141      WARNING api.py:308 -- You are retrieving a sync handle inside an asyncio loop. Try getting client.get_handle(.., sync=False) to get better performance. Learn more at https://docs.ray.io/en/master/serve/http-servehandle.html#sync-and-async-handles component=serve deployment=wrapper replica=wrapper#owpIpe

(ServeController
 pid=1924) 2021-12-15 08:19:50,244      INFO deployment_state.py:940 -- Removing 1 replicas from deployment 'echo'. component=serve deployment=echo

(ServeController
 pid=1924) 2021-12-15 08:19:52,471      INFO deployment_state.py:940 -- Removing 1 replicas from deployment 'wrapper'. component=serve deployment=wrapper

2021-12-15 08:19:54,594 INFO api.py:242 -- Updating deployment 'MultiMethod'. component=serve deployment=MultiMethod
(ServeController
 pid=1924) 2021-12-15 08:19:54,710      INFO deployment_state.py:920 -- Adding 1 replicas to deployment 'MultiMethod'. component=serve deployment=MultiMethod

2021-12-15 08:19:56,255 INFO api.py:249 -- Deployment 'MultiMethod' is ready at `http://127.0.0.1:8000/MultiMethod`. component=serve deployment=MultiMethod
PASSED
::test_repeated_get_handle_cached (ServeController
 pid=1924) 2021-12-15 08:19:56,359      INFO deployment_state.py:940 -- Removing 1 replicas from deployment 'MultiMethod'. component=serve deployment=MultiMethod

2021-12-15 08:19:58,403 INFO api.py:242 -- Updating deployment 'f'. component=serve deployment=f
(ServeController
 pid=1924) 2021-12-15 08:19:58,495      INFO deployment_state.py:920 -- Adding 1 replicas to deployment 'f'. component=serve deployment=f

2021-12-15 08:20:00,845 INFO api.py:249 -- Deployment 'f' is ready at `http://127.0.0.1:8000/f`. component=serve deployment=f
PASSED
::test_args_kwargs[True] (ServeController
 pid=1924) 2021-12-15 08:20:01,228      INFO deployment_state.py:940 -- Removing 1 replicas from deployment 'f'. component=serve deployment=f

2021-12-15 08:20:03,254 INFO api.py:242 -- Updating deployment 'f'. component=serve deployment=f
(ServeController
 pid=1924) 2021-12-15 08:20:03,329      INFO deployment_state.py:920 -- Adding 1 replicas to deployment 'f'. component=serve deployment=f

2021-12-15 08:20:05,896 INFO api.py:249 -- Deployment 'f' is ready at `http://127.0.0.1:8000/f`. component=serve deployment=f
2021-12-15 08:20:05,902 WARNING api.py:308 -- You are retrieving a sync handle inside an asyncio loop. Try getting client.get_handle(.., sync=False) to get better performance. Learn more at https://docs.ray.io/en/master/serve/http-servehandle.html#sync-and-async-handles
PASSED
::test_args_kwargs[False] (ServeController
 pid=1924) 2021-12-15 08:20:06,022      INFO deployment_state.py:940 -- Removing 1 replicas from deployment 'f'. component=serve deployment=f

2021-12-15 08:20:08,101 INFO api.py:242 -- Updating deployment 'f'. component=serve deployment=f
(ServeController
 pid=1924) 2021-12-15 08:20:08,193      INFO deployment_state.py:920 -- Adding 1 replicas to deployment 'f'. component=serve deployment=f

2021-12-15 08:20:10,089 INFO api.py:249 -- Deployment 'f' is ready at `http://127.0.0.1:8000/f`. component=serve deployment=f
PASSED2021-12-15 08:20:10,201   ERROR long_poll.py:155 -- The event loop is closed, shutting down long poll client.

::test_nonexistent_method[True] (ServeController
 pid=1924) 2021-12-15 08:20:10,195      INFO deployment_state.py:940 -- Removing 1 replicas from deployment 'f'. component=serve deployment=f

2021-12-15 08:20:12,309 INFO api.py:242 -- Updating deployment 'A'. component=serve deployment=A
(ServeController
 pid=1924) 2021-12-15 08:20:12,419      INFO deployment_state.py:920 -- Adding 1 replicas to deployment 'A'. component=serve deployment=A

2021-12-15 08:20:14,121 INFO api.py:249 -- Deployment 'A' is ready at `http://127.0.0.1:8000/A`. component=serve deployment=A
2021-12-15 08:20:14,124 WARNING api.py:308 -- You are retrieving a sync handle inside an asyncio loop. Try getting client.get_handle(.., sync=False) to get better performance. Learn more at https://docs.ray.io/en/master/serve/http-servehandle.html#sync-and-async-handles
PASSED
::test_nonexistent_method[False] (ServeController
 pid=1924) 2021-12-15 08:20:14,228      INFO deployment_state.py:940 -- Removing 1 replicas from deployment 'A'. component=serve deployment=A

2021-12-15 08:20:16,353 INFO api.py:242 -- Updating deployment 'A'. component=serve deployment=A
(ServeController
 pid=1924) 2021-12-15 08:20:16,448      INFO deployment_state.py:920 -- Adding 1 replicas to deployment 'A'. component=serve deployment=A

2021-12-15 08:20:17,780 INFO api.py:249 -- Deployment 'A' is ready at `http://127.0.0.1:8000/A`. component=serve deployment=A
PASSED2021-12-15 08:20:17,909   ERROR long_poll.py:155 -- The event loop is closed, shutting down long poll client.

::test_handle_across_loops (ServeController
 pid=1924) 2021-12-15 08:20:17,903      INFO deployment_state.py:940 -- Removing 1 replicas from deployment 'A'. component=serve deployment=A

2021-12-15 08:20:20,021 INFO api.py:242 -- Updating deployment 'A'. component=serve deployment=A
(ServeController
 pid=1924) 2021-12-15 08:20:20,123      INFO deployment_state.py:920 -- Adding 1 replicas to deployment 'A'. component=serve deployment=A

2021-12-15 08:20:21,462 INFO api.py:249 -- Deployment 'A' is ready at `http://127.0.0.1:8000/A`. component=serve deployment=A
PASSED2021-12-15 08:20:21,668   ERROR long_poll.py:155 -- The event loop is closed, shutting down long poll client.
2021-12-15 08:20:21,669 ERROR long_poll.py:155 -- The event loop is closed, shutting down long poll client.
2021-12-15 08:20:21,669 ERROR long_poll.py:155 -- The event loop is closed, shutting down long poll client.
2021-12-15 08:20:21,669 ERROR long_poll.py:155 -- The event loop is closed, shutting down long poll client.
2021-12-15 08:20:21,669 ERROR long_poll.py:155 -- The event loop is closed, shutting down long poll client.
2021-12-15 08:20:21,670 ERROR long_poll.py:155 -- The event loop is closed, shutting down long poll client.
2021-12-15 08:20:21,671 ERROR long_poll.py:155 -- The event loop is closed, shutting down long poll client.
2021-12-15 08:20:21,671 ERROR long_poll.py:155 -- The event loop is closed, shutting down long poll client.
2021-12-15 08:20:21,672 ERROR long_poll.py:155 -- The event loop is closed, shutting down long poll client.
2021-12-15 08:20:21,672 ERROR long_poll.py:155 -- The event loop is closed, shutting down long poll client.
```

What could be causing the timeout?

### Versions / Dependencies

Built latest HEAD with git commit 064f976eb41be854b753fa6aaefa35f3c7b464b7, python3.8.10

### Reproduction script

bazel test python/ray/serve:test_handle

### Are you willing to submit a PR?

- [X] Yes I am willing to submit a PR!

**Comment by pcmoritz** [comment]
I wonder if it goes away if you comment out the `@pytest.mark.asyncio` lines? I have some bad experience with the asyncio plugin of pytest.

**Comment by matthewdeng** [comment]
As a minor note, the [bazel test](https://github.com/ray-project/ray/blob/master/python/ray/serve/BUILD#L159-L160) is configured with a timeout of 60s (from `size = "small"`). I see from the pytest output that it ran in 58.42s, so it might have just passed the timeout threshold in the bazel test.

You can try running `bazel test python/ray/serve:test_handle --test_timeout 120` to confirm that the test finishes?

**Comment by pcmoritz** [comment]
Ah cool, from what @mattip wrote in https://github.com/ray-project/ray/pull/21107 it looks like that fixes it.

What's your recommendation @matthewdeng, should we increase the timeout (and if yes, how -- bump the test to size="medium"?)

**Comment by mattip** [comment]
I moved the test to 'medium' and enabled it in #21107

**Comment by matthewdeng** [comment]
Yeah bumping to `size="medium"` is a reasonable stopgap, though I'm not familiar enough with the test to know if this is expected to take ~60 seconds or if it was caused by a regression which would warrant further investigation.

**Comment by mattip** [comment]
Closing, fixed in #21107
